### DISCUSSION TITLE
Liquid.JS - Pass data to Email body from hook?

### DISCUSSION THREAD

**Discussion by AlexVanK** [directus/directus#14068] [Archive]
I've followed the instructions here 
https://github.com/directus/directus/discussions/12290#discussioncomment-2415816

but no matter what I try, I can't seem to output any of my data in my email body:
![image](https://user-images.githubusercontent.com/56647386/175377736-5ab68e1f-bb03-49bc-aee8-8cff8eaf8493.png)
![image](https://user-images.githubusercontent.com/56647386/175377495-00f677b1-b0ee-4f54-806e-b052b6046f26.png)

and in the template file:
![image](https://user-images.githubusercontent.com/56647386/175377856-44759611-90cb-4950-b3ad-19888863d68f.png)

which produces this:
![image](https://user-images.githubusercontent.com/56647386/175377927-b3e09a3b-d3bb-4a5d-aaa6-0bff8ab37218.png)

Nothing being output.

**Comment by rijkvanzanten** [comment]
@br41nslug could you give this a whirl?

**Reply by keesvanbemmel** [reply]
This works for me:
test.liquid
```
{% layout "base" %}
{% block content %}

<p>
	Someone filled out the contact form. Here are the details.
</p>

<table>
    <tr><td><b>Name:</b></td><td>{{ payload.name }}</td></tr>
    <tr><td><b>Email:</b></td><td>{{ payload.email }}</td></tr>
    <tr><td><b>Company:</b></td><td>{{ payload.company }}</td></tr>
    <tr><td><b>Phone:</b></td><td>{{ payload.phone }}</td></tr>
    <tr><td colspan=2></td></tr>
    <tr><td colspan=2><b>Message:</b></td></tr>
    <tr><td colspan=2>{{ payload.message }}</td></tr>
</table>

<p>
    Thank you,<br>
    {{ projectName }}
</p>

{% endblock %}
```

index.js hook
```
async function sendEmailNotification(input, context) {
  const mailService = new hookcontext.services.MailService({
    schema: context.schema,
    accountability: context.accountability,
  });
  const { payload } = input;
  await mailService.send({
    template: {
      name: 'test',
      data: {
        payload,
      },
    },
    to: ['admin@example.com'],
    subject: 'New Site Contact Submission',
  });
}
```

You need to have access to the MailService, but I'm sure you already have that working for you.

Good luck!

**Reply by br41nslug** [reply]
Been doing some testing on this but i am unable to get the same result. The only way im not getting any data in my templates if it is actually an empty string, null or the variable is non existent any other types of values will have some sort of output 
![image](https://user-images.githubusercontent.com/9389634/176882919-cf4dfe43-f66e-462f-bf64-71c303d6c1f8.png)
Could you check the data object before sending it using a logger?

**Reply by AlexVanK** [reply]
Thank you for your suggestions, I was able to pull data in my email body.
